modules/k8s/job.go

```diff
@@ -0,0 +1,93 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/testing"
+)
+
+// ListJobs will look for Jobs in the given namespace that match the given filters and return them. This will fail the
+// test if there is an error.
+func ListJobs(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) []batchv1.Job {
+	jobs, err := ListJobsE(t, options, filters)
+	require.NoError(t, err)
+	return jobs
+}
+
+// ListJobsE will look for jobs in the given namespace that match the given filters and return them.
+func ListJobsE(t testing.TestingT, options *KubectlOptions, filters metav1.ListOptions) ([]batchv1.Job, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clientset.BatchV1().Jobs(options.Namespace).List(context.Background(), filters)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Items, nil
+}
+
+// GetJob returns a Kubernetes job resource in the provided namespace with the given name. This will
+// fail the test if there is an error.
+func GetJob(t testing.TestingT, options *KubectlOptions, jobName string) *batchv1.Job {
+	job, err := GetJobE(t, options, jobName)
+	require.NoError(t, err)
+	return job
+}
+
+// GetJobE returns a Kubernetes job resource in the provided namespace with the given name.
+func GetJobE(t testing.TestingT, options *KubectlOptions, jobName string) (*batchv1.Job, error) {
+	clientset, err := GetKubernetesClientFromOptionsE(t, options)
+	if err != nil {
+		return nil, err
+	}
+	return clientset.BatchV1().Jobs(options.Namespace).Get(context.Background(), jobName, metav1.GetOptions{})
+}
+
+// WaitUntilJobSucceed waits until job suceeded is superior to 0, retrying the check for the specified amount of times, sleeping
+// for the provided duration between each try. This will fail the test if there is an error or if the check times out.
```

Comment:
```suggestion
// WaitUntilJobSucceed waits until requested job is succeeded, retrying the check for the specified amount of times, sleeping
// for the provided duration between each try. This will fail the test if there is an error or if the check times out.
```